tools/stimulus/types-RC.yaml

```
@@ -39,7 +39,17 @@ INTEGER:
         OUT: '&%C%'
         INOUT: '&%C%'
     INCONV:
-        IN: '%C%=(igraph_integer_t) REAL(%I%)[0];'
+        IN: |-
+            if (Rf_xlength(%I%) == 1) {
+              if (((igraph_integer_t) REAL(%I%)[0]) != REAL(%I%)[0]) {
+                igraph_errorf("The value %.17g is not representable as an integer.",
+                              __FILE__, __LINE__, IGRAPH_EINVAL, REAL(%I%)[0]);
+              }
+              %C% = (igraph_integer_t) REAL(%I%)[0]; 
+            } else {
+              igraph_errorf("Expecting a scalar integer but received a vector of length %zu.", 
+                            __FILE__, __LINE__, IGRAPH_EINVAL, (size_t) Rf_xlength(%I%));
+            }
```

Unhappy paths first:

```suggestion
            if (Rf_xlength(%I%) != 1) {
              igraph_errorf("Expecting a scalar integer but received a vector of length %zu.", 
                            __FILE__, __LINE__, IGRAPH_EINVAL, (size_t) Rf_xlength(%I%));
            }
            if (((igraph_integer_t) REAL(%I%)[0]) != REAL(%I%)[0]) {
              igraph_errorf("The value %.17g is not representable as an integer.",
                            __FILE__, __LINE__, IGRAPH_EINVAL, REAL(%I%)[0]);
            }
            %C% = (igraph_integer_t) REAL(%I%)[0]; 
```

I also wonder if this should use `Rf_error()` instead.